aws-sdk-core/lib/aws-sdk-core.rb

```diff
@@ -266,6 +268,9 @@ module Xml
     autoload :Parser, 'aws-sdk-core/xml/parser'
   end
 
+  # @api private
+  SHARED_CONFIG = SharedConfig.new
+
```

Comment:
I would suggest replacing the constant with a method that memoizes. We had to make a similar change because of how autoload works. It is possible for a second threat to get a no such constant error under concurrency. The code below mitigates this. It is possible that multiple threads hit the method, but a single object will be memoized for later use.

```ruby
def shared_config
  @shared_config ||= SharedConfig.new
end
``